### PR TITLE
Insert data

### PR DESCRIPTION
**Pull request by Hadeel-Salamin**
most changes have been mentioned in the commits

### CODE REVIEW

**Comment by m4v15** [comment]
This contains all the changes from #29 (and #30) so this one has to wait until those have been merged before it can be reviewed.

It's fine to have worked on it all at once, but you should try to make the branches completely independent. To do this, you should have added and committed and pushed the stuff for #29, then checked out to master, then made a new branch (i.e. this one) and then added and committed the stuff here. then we wouldn't have the first 4 commits of this branch and this would be easier to review (as less files changed)

**Review by amanshawar — APPROVED** [review]

**Review by Samaamro20 — APPROVED** [review]
OK well Done

**Review by m4v15 — APPROVED** [review]

**Pull request merged**
